home/export_assessments.py

```diff
@@ -0,0 +1,257 @@
+import copy
+import csv
+from dataclasses import asdict, astuple, dataclass, fields
+from math import ceil
+
+from django.http import HttpResponse  # type: ignore
+from openpyxl.styles import Border, Color, Font, NamedStyle, PatternFill, Side
+from openpyxl.utils import get_column_letter
+from openpyxl.workbook import Workbook
+from openpyxl.worksheet.worksheet import Worksheet
+from wagtail.blocks import (  # type: ignore
+    StreamValue,
+    StructValue,  # type: ignore
+)
+from wagtail.query import PageQuerySet  # type: ignore
+
+
+@dataclass
+class ExportRow:
+    title: str = ""
+    tags: str = ""
+    slug: str = ""
+    locale: str = ""
+    high_result_page: str = ""
+    high_inflection: str = ""
+    medium_result_page: str = ""
+    medium_inflection: str = ""
+    low_result_page: str = ""
+    low_inflection: str = ""
+    generic_error: str = ""
+    question_count: int = 0
+    question: str = ""
+    error: str = ""
+    answers: str = ""
+    scores: str = ""
+
+    @classmethod
+    def headings(cls) -> list[str]:
+        return [f.name for f in fields(cls)]
+
+    def to_dict(self) -> dict[str, str | int]:
+        return asdict(self)
+
+    def to_tuple(self) -> tuple[str | int, ...]:
+        return astuple(self)
+
+
+# Define your AssessmentExporter class
+class AssessmentExporter:
+    rows: list[dict[str | int, str | int]]
+
+    def __init__(self, queryset: PageQuerySet):
+        self.rows = []
+        self.queryset = queryset
+
+    def perform_export(self) -> list[dict[str | int, str | int]]:
+        for item in self.queryset:
+            questions_data = self.get_questions(item.questions)
+            for i, question_data in enumerate(questions_data):
+
+                self.rows.append(
+                    {
+                        "title": item.title,
+                        # "page_id": item.id,
+                        "tags": self._comma_sep_qs(item.tags.all()),
+                        "slug": item.slug,
+                        "locale": str(item.locale.language_code),
+                        "high_result_page": str(item.high_result_page.slug),
+                        "high_inflection": item.high_inflection,
+                        "medium_result_page": str(item.medium_result_page.slug),
+                        "medium_inflection": item.medium_inflection,
+                        "low_result_page": str(item.low_result_page.slug),
+                        "low_inflection": 0,
+                        "generic_error": item.generic_error,
+                        "question_count": i + 1,
```

Comment:
I'm unsure what this `question_count` is? It looks to be like it's the index of the answer for this specific question? In that case, maybe `answer_index` is a better name for it?

It also looks like we're not looking at it at all in the import. So to avoid confusion, I would suggest either removing it, or making the import order the answers for each question by this field.